lib/view_component/compiler.rb

```diff
@@ -125,39 +125,8 @@ def templates
       end
     end
 
-    def matching_views_in_source_location
-      source_location = component_class.source_location
-      return [] unless source_location
-
-      extensions = ActionView::Template.template_handler_extensions.join(",")
-
-      # view files in a directory named like the component
-      directory = File.dirname(source_location)
-      filename = File.basename(source_location, ".rb")
-      component_name = component_class.name.demodulize.underscore
-
-      # Add support for nested components defined in the same file.
-      #
-      # e.g.
-      #
-      # class MyComponent < ViewComponent::Base
-      #   class MyOtherComponent < ViewComponent::Base
-      #   end
-      # end
-      #
-      # Without this, `MyOtherComponent` will not look for `my_component/my_other_component.html.erb`
-      nested_component_files = if component_class.name.include?("::") && component_name != filename
-        Dir["#{directory}/#{filename}/#{component_name}.*{#{extensions}}"]
-      else
-        []
-      end
-
-      # view files in the same directory as the component
-      sidecar_files = Dir["#{directory}/#{component_name}.*{#{extensions}}"]
-
-      sidecar_directory_files = Dir["#{directory}/#{component_name}/#{filename}.*{#{extensions}}"]
-
-      (sidecar_files - [source_location] + sidecar_directory_files + nested_component_files)
+    def template_paths
```

Comment:
Thoughts on inlining this call in `#templates` since it looks like that's the only place it's used?